mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -193,6 +200,44 @@ uploadSingle file a = do
   x <- liftIO $ LBS.readFile file
   void . send $ fencode' putObject a (toBody x) & poServerSideEncryption .~ pure sse
 
+-- | Upload a file from a URL to a given S3 Address
+uploadFromURL :: URL -> Address -> AWS UploadResult
+uploadFromURL = uploadFromURLWithMode Fail
+
+-- | Upload a file from a URL to a given S3 Address
+--   fail if the destination already exists and the WriteMode is Fail
+uploadFromURLWithMode :: WriteMode -> URL -> Address -> AWS UploadResult
+uploadFromURLWithMode m url a = eitherT (pure . UploadError) (const $ pure UploadOk) $ do
+  when (m == Fail) . whenM (lift $ exists a) . left $ UploadDestinationExists a
+  EitherT (Right <$> do
+    manager  <- liftIO $ newManager tlsManagerSettings
```

Comment:
This needs to be passed in not created, it is really expensive. Or actually you can just use the one from AWS, call getEnv and you can get access to the manager, save creating one at all.